### PR TITLE
task 2

### PR DESCRIPTION
**Pull request by AKhlebko**
Anton Khlebko task 2

### CODE REVIEW

File: DEV-2/DEV-2/OnlyEvenPositionCharsString.cs [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+            if (processedString.Length < 2)
+            {
+                return;
+            }
+            int deletePos = 1;
+            while (deletePos < processedString.Length)
+            {
+                processedString.Remove(deletePos, 1);
+                deletePos++;
+            }
```
Comment (IgorKheidorov) [inline DEV-2/DEV-2/OnlyEvenPositionCharsString.cs]: Single responsibility principle violation. One method- one activity

File: DEV-2/DEV-2/OnlyEvenPositionCharsString.cs [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+
+namespace TaskDEV2
+{
+    /// <summary>
+    /// Class contains methods for deleting all odd symbols 
+    /// in the string using StringBilder
+    /// </summary>
+    class OnlyEvenPositionCharsString
+    {
+        StringBuilder processedString;
```
Comment (IgorKheidorov) [inline DEV-2/DEV-2/OnlyEvenPositionCharsString.cs]: bad name

File: DEV-2/DEV-2/OnlyEvenPositionCharsString.cs [hunk 3]
```diff
@@ -0,0 +1,45 @@
...
+        {
+            processedString = str;
+        }
+        
+        /// <summary>
+        /// Method for leaves only even positions char
+        /// in the processedString
+        /// </summary>
+        public void LeaveOnlyEvenPositionChars()
+        {
```
Comment (IgorKheidorov) [inline DEV-2/DEV-2/OnlyEvenPositionCharsString.cs]: Why? if length is 1 you have to print the string, 0 index we assume even

File: DEV-2/DEV-2/OnlyEvenPositionCharsString.cs [hunk 4]
```diff
@@ -0,0 +1,45 @@
...
+            if (processedString.Length < 2)
+            {
+                return;
+            }
+            int deletePos = 1;
+            while (deletePos < processedString.Length)
```
Comment (IgorKheidorov) [inline DEV-2/DEV-2/OnlyEvenPositionCharsString.cs]: I seems that you delete each symbol, unclear how it works

File: DEV-2/DEV-2/EntryPoint.cs [hunk 5]
```diff
@@ -0,0 +1,15 @@
+﻿using System;
+using System.Text;
+
+namespace TaskDEV2
+{
+    class EntryPoint
+    {
+        static void Main(string[] args)
+        {
+            StringBuilder str = new StringBuilder(Console.ReadLine());
```
Comment (IgorKheidorov) [inline DEV-2/DEV-2/EntryPoint.cs]: How user can understand that he has to input something?

File: DEV-2/DEV-2/EntryPoint.cs [hunk 6]
```diff
@@ -0,0 +1,22 @@
...
+
+namespace TaskDEV2
+{
+    class EntryPoint
+    {
+        static void Main(string[] args)
+        { 
+            string inputString;
+            Console.Write("Input string to delete odd positions chars: ");
+            inputString = Console.ReadLine();
```
Comment (AlexanderSoroka) [inline DEV-2/DEV-2/EntryPoint.cs]: Declare variable as close to first usage as possible

File: DEV-2/DEV-2/OnlyEvenPositionCharsString.cs [hunk 7]
```diff
@@ -0,0 +1,41 @@
+﻿using System;
+using System.Text;
+
+namespace TaskDEV2
+{
+    /// <summary>
+    /// Class contains methods for deleting all odd symbols 
+    /// in the string using StringBilder
+    /// </summary>
+    class OnlyEvenPositionCharsString
```
Comment (AlexanderSoroka) [inline DEV-2/DEV-2/OnlyEvenPositionCharsString.cs]: Bad class name

File: DEV-2/DEV-2/OnlyEvenPositionCharsString.cs [hunk 8]
```diff
@@ -0,0 +1,41 @@
...
+        /// processedStringBuilder which are on even indexes
+        /// </summary>
+        public void PrintOnlyEvenPositionChars()
+        {
+            StringBuilder outputEvenString = new StringBuilder();
+            for (int evenPos = 0; evenPos < processedStringBuilder.Length; evenPos += 2)
+            {
+                outputEvenString.Append(processedStringBuilder[evenPos]);
+            }
+            Console.WriteLine($"String with only even position chars - {outputEvenString}");
```
Comment (AlexanderSoroka) [inline DEV-2/DEV-2/OnlyEvenPositionCharsString.cs]: single responsibility violation

File: DEV-2/DEV-2/OnlyEvenPositionCharsString.cs [hunk 9]
```diff
@@ -0,0 +1,52 @@
...
+    /// <summary>
+    /// Class contains methods for deleting all odd symbols 
+    /// in the string using StringBilder
+    /// </summary>
+    class OddIndexCharsDeleter
+    {
+        StringBuilder processedStringBuilder;
+
+        /// <summary>
+        /// Initializer for class's instances
```
Comment (AlexanderSoroka) [inline DEV-2/DEV-2/OnlyEvenPositionCharsString.cs]: Constructor not Initializer

File: DEV-2/DEV-2/OnlyEvenPositionCharsString.cs [hunk 10]
```diff
@@ -0,0 +1,52 @@
...
+        {
+            processedStringBuilder = new StringBuilder(str);
+        }
+
+        /// <summary>
+        /// Method prints string builder from GetEvenInderStringBuilder 
+        /// </summary>
+        public void PrintOnlyEvenPositionChars()
+        {
+            Console.WriteLine($"String with only even position chars - {GetEvenIndexStringBulder()}");
```
Comment (AlexanderSoroka) [inline DEV-2/DEV-2/OnlyEvenPositionCharsString.cs]: single responsibility violation

File: DEV-2/DEV-2/OnlyEvenPositionCharsString.cs [hunk 11]
```diff
@@ -0,0 +1,52 @@
...
+        }
+
+        /// <summary>
```
Comment (AlexanderSoroka) [inline DEV-2/DEV-2/OnlyEvenPositionCharsString.cs]: do not need to comment private methods

File: DEV-2/DEV-2/OnlyEvenPositionCharsString.cs [hunk 12]
```diff
@@ -0,0 +1,52 @@
...
+        /// Makes StringBuilder made of chars from
+        /// processedStringBuilder which are on even indexes
+        /// </summary>
+        /// <returns>
+        /// Returns made StringBuilder
+        /// </returns>
+        private StringBuilder GetEvenIndexStringBulder()
+        {
+            StringBuilder outputEvenStringBuilder = new StringBuilder();
+            for (int evenPos = 0; evenPos < processedStringBuilder.Length; evenPos += 2)
```
Comment (AlexanderSoroka) [inline DEV-2/DEV-2/OnlyEvenPositionCharsString.cs]: just i would be enough

**Pull request merged**